File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int answer = 1;
        int start = 1;

        while (start <= n / 2) {
            int sum = 0;
            for (int i = start; i <= n / 2 + 1; i++) {
                sum += i;
                if (sum == n) {
                    answer++;
                    break;
                } else if (sum > n) {
                    break;
                }
            }
            start++;
        }
        return answer;
    }
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    answer = 0
    start = 1

    while start <= n:
        s = 0
        for i in range(start, n + 1):
            s += i
            if s == n:
                answer += 1
                break
            elif s > n:
                break
        
        start += 1
    return answer